File: eDentist/eDentist.Services/Database/Appointment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eDentist.Services.Database
{
    public class Appointment
    {
        [Key]
        public int AppointmentID { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; }
        public int UserId { get; set; }
        public DateTime Datum { get; set; }
        public string Vrijeme { get; set; }
        public bool TerminZatvoren { get; set; }

        [ForeignKey("UserIdDentist")]
        public User? UserDentist { get; set; }
        public int? UserIdDentist { get; set; }

    }
}

File: eDentist/UI/edentist_admin/lib/models/appointment.dart
import 'package:json_annotation/json_annotation.dart';

part "appointment.g.dart";

@JsonSerializable()
class Appointment {
  int? appointmentID;
  int? userId;
  DateTime datum;
  String? vrijeme;
  bool terminZatvoren;
  int? userIdDentist;
  Appointment(this.appointmentID, this.userId, this.datum, this.vrijeme,
      this.terminZatvoren, this.userIdDentist);

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);
  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
